src/backend/base/langflow/custom/custom_component/component.py

```diff
@@ -506,11 +507,10 @@ def __call__(self, **kwargs):
     async def _run(self):
         # Resolve callable inputs
         for key, _input in self._inputs.items():
-            if callable(_input.value):
-                result = _input.value()
-                if inspect.iscoroutine(result):
-                    result = await result
-                self._inputs[key].value = result
+            if asyncio.iscoroutinefunction(_input.value):
+                self._inputs[key].value = await _input.value()
+            elif callable(_input.value):
+                self._inputs[key].value = await asyncio.to_thread(_input.value())
```

Comment:
It seems if the input value is a standard callable, there is a risk that it makes blocking operations.

Reply:
Should we be calling _input.value inside `to_thread` or passing the callable?
```suggestion
                self._inputs[key].value = await asyncio.to_thread(_input.value)
```

Reply:
Good catch, thanks!